Tests/JobsTests/JobsTests.swift

```diff
@@ -137,6 +137,19 @@ final class JobsTests: XCTestCase {
         XCTAssertEqual(failedJobCount.load(ordering: .relaxed), 1)
     }
 
+    func testDelayedJob() async throws {
+        let jobIdentifer = JobIdentifier<Int>(#function)
+        let expectation = XCTestExpectation(description: "TestJob.execute was called")
+        let jobQueue = JobQueue(.memory, numWorkers: 1, logger: Logger(label: "JobsTests"))
+        jobQueue.registerJob(id: jobIdentifer) { _, _ in
+            expectation.fulfill()
+        }
+        try await self.testJobQueue(jobQueue) {
+            try await jobQueue.push(id: jobIdentifer, parameters: 0, delayedUntil: .now.addingTimeInterval(5))
+            await fulfillment(of: [expectation], timeout: 10)
+        }
```

Comment:
This isn't really testing the job was delayed. It only tests it runs. Perhaps you should run two jobs, the first one is delayed and the second one isn't. Then make sure the second one ran first.

Reply:
I will add another job without any delayed and check the execution order.

Reply:
The test has been updated to confirm that a delayed job is executed after a non delayed job.